src/dynamics/video_player/player/player.js

```diff
@@ -2075,6 +2075,8 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                 },
 
                 hidePlayerContainer: function() {
+                    // If outstream and we need to repeat then ignore hidePlayer
+                    if (this.get("outstream") && !this.get("outstreamoptions.hideOnCompletion")) return;
                     // If no there will be "states.hiddenelement.visible" condition, states will be overwritten
```

Comment:
Hi @sambua, this seems a bit counterintuitive, I would expect calling `hidePlayerContainer` to always hide the player. Having to set `hideOnCompletion` to true to make it work doesn't seem right to me, it might be clearer to handle this condition outside of the `hidePlayerContainer` method, ensuring that this method's responsibility is solely to hide the player.
```suggestion
```
Let me know your thoughts on this.

Reply:
Hi @ntshcalleia sure, I understood your point. But this line will ensure that calling hide player in somewhere by fault, when hideOnCompletion was set as false, will not close outstream player container. 

Currently ad-error event trigger to close it and this line disallow this operation, as repeat button will re-create it. We also need to hide player on ad:error and call if we have some fallback urls after some period of time.